addons/website_livechat/static/src/models/visitor/visitor.js

```diff
@@ -0,0 +1,164 @@
+odoo.define('website_livechat/static/src/models/partner/partner.js', function (require) {
+'use strict';
+
+const { registerNewModel } = require('mail/static/src/model/model_core.js');
+const { attr, many2one, one2many } = require('mail/static/src/model/model_field.js');
+
+function factory(dependencies) {
+
+    class Visitor extends dependencies['mail.model'] {
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('country_id' in data && data.country_id) {
+                data2.country = [['insert', {
+                    id: data.country_id,
+                    code: data.country_code,
+                }]];
+            }
```

Comment:
For consistency `else` (falsy data) should do `unlink` just like for `partner_id`.
(Unless the server is doing stupid stuff of wrongly setting the value to `False` sometimes? But then it should be commented.)